TinYard/Extensions/CallbackTimer/Impl/Services/CallbackTimerService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using TinYard.API.Interfaces;
+using TinYard.Extensions.CallbackTimer.API.Services;
+using TinYard.Extensions.CallbackTimer.Impl.VO;
+using TinYard.Framework.Impl.Attributes;
+
+namespace TinYard.Extensions.CallbackTimer.Impl.Services
+{
+    public class CallbackTimerService : ICallbackTimer
+    {
+        [Inject]
+        public IContext context;
+
+        private System.Threading.CancellationTokenSource _updateLoopCancelToken;
+
+        private List<Timer> _timers = new List<Timer>();
+
+        public CallbackTimerService()
+        {
+            //TODO : Add in when added to IContext
+            //context?.OnDestroy += OnDestroy;
+
+            _timers = new List<Timer>();
+            _updateLoopCancelToken = new System.Threading.CancellationTokenSource();
+
+            Task.Run(Update, _updateLoopCancelToken.Token);
+        }
+
+        ~CallbackTimerService()
```

Comment:
IMO OnDestroy should be called manually in C# due to the garbage collector and not knowing exactly when this object will be collected. While I'm sure this won't change anything (as I assume this class has no dependencies on it so when it's cleaned up doesn't matter) from my experience it's good to know when things are destroyed.

(And this comment feels like I'm back at Leamington doing code reviews again)

Reply:
Usually I would agree, and in all honesty I'm not sure if this deconstructor is needed.

I've added it in as we're doing an infinite running Task and I wanted to make sure it's cancelled - Not my usual area though so I'm unsure if this is even needed?